Client/Frontend/Popup/PaymentHandlerPopupView.swift

```diff
@@ -0,0 +1,284 @@
+        totalItem.textAlignment = .left
+        totalItem.font = UIFont.systemFont(ofSize: 12, weight: UIFont.Weight.regular)
+        totalItem.text = message
+        totalItem.numberOfLines = 0
+        
+        containerView.addSubview(totalItem)
+        updateSubviews()
+    }
+    
+    func clearDisplayItems() {
+        displayItems.removeAll()
+    }
+    
+    func update(title: String) {
+        titleLabel.text = title
+    }
+    
+    func clearTextField() {
+        textField?.text = nil
+    }
+    
+    func updateSubviews() {
+        titleLabel.adjustsFontSizeToFitWidth = false
+        messageLabel.adjustsFontSizeToFitWidth = false
+        displayItemTitle.adjustsFontSizeToFitWidth = false
+        totalItemTitle.adjustsFontSizeToFitWidth = false
+        
+        for label in displayItems {
+            label.adjustsFontSizeToFitWidth = false
+        }
+
+        updateSubviews(resizePercentage: 1.0)
+        
+        let paddingHeight = padding * 3.0
+        let externalContentHeight = dialogButtons.count == 0 ? paddingHeight : kPopupDialogButtonHeight + paddingHeight
+        let desiredHeight = UIScreen.main.bounds.height - externalContentHeight
+        
+        if containerView.frame.height > desiredHeight {
+            let resizePercentage = desiredHeight / containerView.frame.height
+            titleLabel.adjustsFontSizeToFitWidth = true
+            messageLabel.adjustsFontSizeToFitWidth = true
+            displayItemTitle.adjustsFontSizeToFitWidth = true
+            totalItemTitle.adjustsFontSizeToFitWidth = true
+            for label in displayItems {
+                label.adjustsFontSizeToFitWidth = true
+            }
+            updateSubviews(resizePercentage: resizePercentage)
+        }
+        
+        // Lottie animation stops playing when view is not visible, the animation needs to be resumed.
+        if let animationView = dialogImage as? AnimationView, !animationView.isAnimationPlaying {
+            animationView.play()
+        }
+    }
+    
+    fileprivate func updateSubviews(resizePercentage: CGFloat) {
+        let width: CGFloat = dialogWidth
+        
+        var imageFrame: CGRect = dialogImage?.frame ?? CGRect.zero
+        if let dialogImage = dialogImage {
+            let dialogImageSize = dialogImage.frame.size
+            imageFrame.size = CGSize(width: dialogImageSize.width * resizePercentage, height: dialogImageSize.height * resizePercentage)
+            imageFrame.origin.x = (width - imageFrame.width) / 2.0
+            imageFrame.origin.y = kPadding * 2.0 * resizePercentage
+            dialogImage.frame = imageFrame
+        }
+        
+        var titleLabelSize: CGSize = titleLabel.sizeThatFits(CGSize(width: width - kPadding * 3.0, height: CGFloat.greatestFiniteMagnitude))
+        titleLabelSize.height = titleLabelSize.height * resizePercentage
+        var titleLabelFrame: CGRect = titleLabel.frame
+        titleLabelFrame.size = titleLabelSize
```

Comment:
These should be updated with constraints, we shouldn't use size, origin, frame, at all. If you need help converting this to use snapkit, then ping #iOS. 

`titleLabel.snp.makeConstraints...`

Reply:
UI will be dumped.